File: measure.rb
if ARGV.size < 1
  print("Provide filename\n")
  exit(-1)
end

k = 256
$numOfNums = k * 1024 * 1024
$fileName = ARGV[0]
$reps = 6

def writeUsingFile()
  open($fileName, 'wb') do |writer| 
    (1..$numOfNums).each { |i| writer.putc(i & 0xff) }
  end
end

def readUsingFile()
  open($fileName, 'rb') do |reader| 
    (1..$numOfNums).each { reader.getc() }
  end
  File.delete($fileName)
end

if File.exists?($fileName)
  File.delete($fileName)
end
(read, write) = (1..$reps).collect { 
  time1 = Process.clock_gettime(Process::CLOCK_MONOTONIC, :nanosecond)
  writeUsingFile()
  time2 = Process.clock_gettime(Process::CLOCK_MONOTONIC, :nanosecond)
  readUsingFile()
  time3 = Process.clock_gettime(Process::CLOCK_MONOTONIC, :nanosecond)
  tmp1 = $numOfNums.to_f / 1024 / 1024 * 1e9
  readTime = tmp1 / (time3 - time2)
  writeTime = tmp1 / (time2 - time1)
  print("#{readTime}, #{writeTime}\n")
  STDOUT.flush
  [readTime, writeTime]
}[1..$reps].inject([0, 0]) { |x, y| [x[0] + y[0], x[1] + y[1]] }
tmp1 = $reps - 1
print "Bytes: #{$numOfNums / 1024 / 1024} MiB, ReadRate: #{read / tmp1} MB/s, WriteRate: #{write / tmp1} MB/s"

File: measure.py
import os, sys, time

if len(sys.argv) < 2:
    print("Provide filename")
    sys.exit(-1)

k = 256
numOfNums = k * 1024 * 1024
fileName = sys.argv[1]
reps = 6

def writeUsingFile():
  with open(fileName, 'w', encoding='iso8859') as writer: 
    for i in range(1, numOfNums):
      writer.write(chr(i & 0xff))

def readUsingFile():
  with open(fileName, 'r', encoding='iso8859') as reader: 
    for i in range(1, numOfNums):
      reader.read(1)
  os.remove(fileName)

def foo(x):
  time1 = time.perf_counter()
  writeUsingFile()
  time2 = time.perf_counter()
  readUsingFile()
  time3 = time.perf_counter()
  tmp1 = numOfNums / 1024 / 1024 
  readTime = tmp1 / (time3 - time2)
  writeTime = tmp1 / (time2 - time1)
  print("{0} {1}".format(readTime, writeTime))
  sys.stdout.flush()
  return [readTime, writeTime]

if os.path.exists(fileName):
  os.remove(fileName)
tmp1 = [foo(x) for x in range(1, reps + 1)]
read, write = map(sum, zip(*(tmp1[1:])))
tmp2 = reps - 1
print("Bytes: {0} MiB, ReadRate: {1} MB/s, WriteRate: {2}".format(numOfNums / 1024 / 1024, read / tmp2, write / tmp2))
